src/inverseRobot.js

```diff
@@ -1,13 +1,31 @@
 'use strict';
 
-/*
+/**
  * @param {object} robot
  *
- * @return {object}
-*/
+ * @return {object | null}
+ */
 
 function inverseRobot(robot) {
-  // write code here
+  const invertedRobotMap = new Map();
+  const valuesArray = [];
+
+  for (const key in robot) {
+    if (robot.hasOwnProperty(key)) {
```

Comment:
it has no sense, key would definitely be in robot, because you iterate through robot😉

Reply:
I agree, you want to check `invertedRobot.hasOwnProperty()` to check for duplicates. 
`robot.hasOwnProperty(key)`  always resolves to true.